tools/cgdiff/include/NodeSummaryHasher.h

```diff
@@ -0,0 +1,50 @@
+/**
+ * File: NodeSummaryHasher.h 
+ * License: Part of the MetaCG project. Licensed under BSD 3 clause license. See LICENSE.txt file at
+ * https://github.com/tudasc/metacg/LICENSE.txt
+ */
+
+#pragma once
+#include "NodeSummary.h"
+
+/*
+ * Provides a custom hash function for NodeSummary objects that respects
+ * a ComparisonMode. Can be used with unordered containers (e.g., std::unordered_set)
+ * to compare nodes based on specific fields while optionally ignoring others.
+ *
+ * The hash is order-independent for sets (callees, metadata) using XOR,
+ * and combines individual field hashes using boost's hash-combine.
+ */
+struct NodeSummaryHasher {
```

Comment:
```suggestion
class NodeSummaryHasher {
```
If we use `public` and `private`, I think we should use `class`.